gap/attributes/attrinv.gi

```diff
@@ -36,25 +36,32 @@ function(S)
   if not IsFinite(S) then
     ErrorNoReturn("Semigroups: NaturalPartialOrder: usage,\n",
                   "the argument is not a finite semigroup,");
-  fi;
-
-  if not IsInverseSemigroup(S) then
+  elif not IsInverseSemigroup(S) then
     ErrorNoReturn("Semigroups: NaturalPartialOrder: usage,\n",
                   "the argument is not an inverse semigroup,");
+  elif IsPartialPermSemigroup(S) then
+    # Use the library method for partial perm inverse semigroups.
+    return NaturalPartialOrder(S);
   fi;
 
   Info(InfoWarning, 2, "NaturalPartialOrder: this method ",
                        "fully enumerates its argument!");
 
   elts := ShallowCopy(Elements(S));
-  p    := Sortex(elts, IsGreensDGreaterThanFunc(S)) ^ -1;
+  func := function(x, y)
+    return IsGreensDGreaterThanFunc(S)(y, x);
+  end;
```

Comment:
Can you use a 2-argument lambda here instead? `p := Sortex(elts, {x, y} -> IsGreensDGreaterThanFunc(S)(y, x));`?

Reply:
Save some lines.